### PR TITLE
Update Wasmtime to v0.28

### LINKED ISSUE #70: Update Wasmtime to v0.28
Note that there are quite a few breaking changes in the new Wasmtime API.

### PR DESCRIPTION
**Pull request by radu-matei**
closes #70

### CODE REVIEW

File: tests/integration.rs [hunk 1]
```diff
@@ -2,10 +2,11 @@
 mod tests {
     use anyhow::Error;
     use std::time::Instant;
-    use wasi_cap_std_sync::WasiCtxBuilder;
+    // use wasi_cap_std_sync::WasiCtxBuilder;
```
Comment (bacongobbler) [inline tests/integration.rs:5]: can this be removed, or was this intentionally stubbed out?

**Review by bacongobbler — APPROVED** [review]
Code LGTM, though someone else more familiar with the project may be more qualified to verify the behavioral differences between 0.26 and 0.28.

File: crates/wasi-experimental-http-wasmtime/src/lib.rs [hunk 2]
```diff
@@ -91,88 +114,101 @@ impl HostCalls {
...
     ) -> Result<(), HttpError> {
-        let mut st = st.borrow_mut();
-        // Get the current response body.
-        let mut body = &mut st
-            .responses
-            .get_mut(&handle)
-            .ok_or(HttpError::InvalidHandle(handle))?
-            .body;
-        let memory = memory_get(caller)?;
+        let mut st = st.write()?;
```
Comment (thomastaylor312) [inline crates/wasi-experimental-http-wasmtime/src/lib.rs:134]: For what it is worth, I would just panic if poison occurs (which is very very rare, in fact, they are thinking of removing it). Particularly in light of this info from the documentation:

> The precise semantics for when a lock is poisoned is documented on each lock, but once a lock is poisoned then all future acquisitions will return this error.

So if it does get poisoned you are SOL

**Review by thomastaylor312 — APPROVED** [review]

File: crates/wasi-experimental-http-wasmtime/src/lib.rs (hunk 2 above)
Comment (radu-matei) [reply crates/wasi-experimental-http-wasmtime/src/lib.rs:134]: Yeah, but this way we get a chance to return an error to the client.

Comment (radu-matei) [reply crates/wasi-experimental-http-wasmtime/src/lib.rs:134]: I would avoid a panic in the library for a number of reasons, the most important of which being that the runtime could be executing other modules as well.
On top of that, this also gives us a way of returning an error to the client.

Comment (thomastaylor312) [reply crates/wasi-experimental-http-wasmtime/src/lib.rs:134]: As long as that terminates things, I think that is great

**Pull request merged**